tests/functional/aws-node-sdk/test/object/put.js

```diff
@@ -85,6 +85,28 @@ describe('PUT object', () => {
                 });
             });
 
+        it('should return error if putting object w/ > 2KB user-defined md',
+            done => {
+                const metadata = {};
+                const counter = 8;
+                // 256 * 8 = 2048 (2 KB)
+                for (let i = 0; i < counter; i++) {
+                    const key = `header${i}`;
+                    const valueLength =
+                        256 - ('x-amz-meta-'.length + key.length);
```

Comment:
if you can use the constants number rather than hard-coding here that would be good as well